katdal/h5datav2_5.py

```diff
@@ -0,0 +1,631 @@
+"""Data accessor class for HDF5 files produced by AVN DBE."""
+
+import logging
+
+import numpy as np
+import h5py
+import katpoint
+
+from .dataset import DataSet, WrongVersion, BrokenFile, Subarray, \
+    DEFAULT_SENSOR_PROPS, DEFAULT_VIRTUAL_SENSORS, _robust_target
+from .spectral_window import SpectralWindow
+from .sensordata import RecordSensorData, SensorCache
+from .categorical import CategoricalData, sensor_to_categorical
+from .lazy_indexer import LazyIndexer, LazyTransform
+
+logger = logging.getLogger(__name__)
+
+# Simplify the scan activities to derive the basic state of the antenna (slewing, scanning, tracking, stopped)
+SIMPLIFY_STATE = {'scan_ready': 'slew', 'scan': 'scan', 'scan_complete': 'scan', 'track': 'track', 'slew': 'slew'}
+
+SENSOR_PROPS = dict(DEFAULT_SENSOR_PROPS)
+SENSOR_PROPS.update({
+    '*activity': {'greedy_values': ('slew', 'stop'), 'initial_value': 'slew',
+                  'transform': lambda act: SIMPLIFY_STATE.get(act, 'stop')},
+    '*target': {'initial_value': '', 'transform': _robust_target},
+    # These float sensors are actually categorical by nature as they represent user settings
+    'RFE/center-frequency-hz': {'categorical': True},
+    'RFE/rfe7.lo1.frequency': {'categorical': True},
+    '*attenuation': {'categorical': True},
+    '*attenuator.horizontal': {'categorical': True},
+    '*attenuator.vertical': {'categorical': True},
+    'RFE/rfe.band.select.LCP': {'categorical': True},
+    'RFE/rfe.band.select.RCP': {'categorical': True},
+    'RFE/rfe.lo-intermediate.5GHz.frequency': {'categorical': True},
+    'RFE/rfe.lo-intermediate.6_7GHz.frequency': {'categorical': True},
+    'RFE/rfe.lo-final.frequency': {'categorical': True}
+})
+
+SENSOR_ALIASES = {
+    'nd_roach': 'roach.noise.diode.on',
+}
+
+def _calc_azel(cache, name, ant="ant1"):
+    """Calculate virtual (az, el) sensors from actual ones in sensor cache."""
+    real_sensor = 'Antennas/%s/%s' % (ant, 'pos.actual-scan-azim' if name.endswith('az') else 'pos.actual-scan-elev')
+    cache[name] = sensor_data = katpoint.deg2rad(cache.get(real_sensor))
+    return sensor_data
+
+VIRTUAL_SENSORS = dict(DEFAULT_VIRTUAL_SENSORS)
+VIRTUAL_SENSORS.update({'Antennas/{ant}/az': _calc_azel, 'Antennas/{ant}/el': _calc_azel})
+
+FLAG_NAMES = ('reserved0', 'static', 'cam', 'reserved3', 'detected_rfi', 'predicted_rfi', 'reserved6', 'reserved7')
```

Comment:
The flags have moved to a separate `.flags` module but you could always clear this up afterwards if you wish.

Reply:
We don't actually need flags, that was just something that was in h5datav2.py which I copied previously. I will remove the line, I don't think I use it.

Reply:
Then also remove anything mentioning flags. The default `DataSet.flags` property will raise `NotImplementedError` so as long as you don't access it you should be fine.

The same goes for any mentions for weights...

Reply:
Done, commit 04bc711.